08-5-ex-sol-ADHD.Rmd

```diff
@@ -0,0 +1,166 @@
+
+# Calculates Euclidean distances between samples.
+euclidean_dist_relA <- vegan::vegdist(relabundance_assay, method = "euclidean")
+
+# Performing PCoA
+euclidean_relA_pcoa <- ecodist::pco(euclidean_dist_relA)
+
+# Creates a data frame from principal coordinates
+euclidean_relA_pcoa_df <- data.frame(pcoa1 = euclidean_relA_pcoa$vectors[,1], 
+                                pcoa2 = euclidean_relA_pcoa$vectors[,2])
+
+# Visualizing
+library(ggplot2)
+euclidean_relA_plot <- ggplot(data = euclidean_relA_pcoa_df, aes(x=pcoa1, y=pcoa2)) +
+  geom_point() +
+  labs(x = "Principal Component 1",
+       y = "Principal Component 2",
+       title = "Euclidean PCoA with Relative Abundance transformation") +
+  theme(title = element_text(size = 12)) # makes titles smaller
+
+euclidean_relA_plot
+```
+
+ * Investigate the influence of the data transformations on
+   statistical analysis: Visualize community variation with PCoA with
+   the following options: 1) Bray-Curtis distances for compositional
+   data; 2) Euclidean distances for CLR-transformed data.
+
+```{r}
+ # Computing Bray-Curtis distances
+bray_dist_relA <- vegan::vegdist(relabundance_assay, method = "bray")
+bray_dist_relA <- t(bray_dist_relA)
+
+# Computing clr data and its euclidean distance
+tse <- transformCounts(tse, method = "clr", pseudocount = 1)
+clr_assay <- assays(tse)$clr
+euclidean_dist_clr <- vegan::vegdist(clr_assay, method = "euclidean")
+euclidean_dist_clr <- t(euclidean_dist_clr)
+
+# Performing PCoA and making the dataframes
+bray_dist_relA_pcoa <- ecodist::pco(bray_dist_relA)
+euclidean_dist_clr_pcoa <- ecodist::pco(euclidean_dist_clr)
+
+bray_dist_relA_pcoa_df <- data.frame(pcoa1 = bray_dist_relA_pcoa$vectors[,1], 
+                                pcoa2 = bray_dist_relA_pcoa$vectors[,2])
+euclidean_dist_clr_pcoa_df <- data.frame(pcoa1 = euclidean_dist_clr_pcoa$vectors[,1], 
+                                pcoa2 = euclidean_dist_clr_pcoa$vectors[,2])
+
+# Visualizing
+ggplot(data = bray_dist_relA_pcoa_df, aes(x=pcoa1, y=pcoa2)) +
+  geom_point() +
+  labs(x = "PC 1",
+       y = "PC 2",
+       title = "Bray-Curtis PCoA with Relative Abundance transformation") +
+  theme(title = element_text(size = 12))
+ 
+ggplot(data = euclidean_dist_clr_pcoa_df, aes(x=pcoa1, y=pcoa2)) +
+  geom_point() +
+  labs(x = "PC 1",
+       y = "PC 2",
+       title = "Euclidean PCoA with clr transformation") +
+  theme(title = element_text(size = 12))
+```
+
+ * Community-level comparisons: Use PERMANOVA to investigate whether
+   the community composition differs between two groups of individuals
+   (e.g. males and females, or some other grouping of your
+   choice). You can also include covariates such as age and gender,
+   and see how this affects the results.
+   
+```{r}
+permanova_cohort <- vegan::adonis(relabundance_assay ~ cohort,
+                                  data = colData(tse),
+                                  permutations = 9999)
```

Comment:
For the sake of demonstration here, 99 permutations should be enough. let us reduce CO2 emissions... You can mention in a comment that we recommend more for publications (999 or 9999)